packages/headless/src/auth/saml-provider.ts

```diff
@@ -0,0 +1,112 @@
+import {fetch} from 'cross-fetch';
+import {getIsomorphicHistory, IsomorphicHistory} from './isomorphic-history';
+import {getIsomorphicLocation, IsomorphicLocation} from './isomorphic-location';
+
+export interface SamlProviderOptions {
+  organizationId: string;
+  provider: string;
+  request?: Fetch;
+  location?: IsomorphicLocation;
+  history?: IsomorphicHistory;
+}
+
+export interface SamlProvider {
+  exchangeHandshakeToken(): Promise<string>;
+  handshakeTokenAvailable: boolean;
+  login(): void;
+}
+
+type Fetch = (
+  input: RequestInfo,
+  init?: RequestInit | undefined
+) => Promise<Response>;
+
+const handshakeTokenParamName = 'handshake_token';
+
+export function buildSamlProvider(config: SamlProviderOptions): SamlProvider {
+  const options = buildOptions(config);
+  const api = 'https://platform.cloud.coveo.com/rest/search';
```

Comment:
should this be configurable?

Reply:
Yes, I have a [todo ](https://github.com/coveo/ui-kit/pull/1506/files#r754440763)to support adjusting the value,